pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -853,25 +853,29 @@ func (r *ReconcileMachinePool) syncMachineAutoscalers(
 						maLog.WithField("desired", minReplicas).
 							WithField("observed", rMA.Spec.MinReplicas).
 							Info("min replicas out of sync")
-						rMA.Spec.MinReplicas = minReplicas
+						remoteMachineAutoscalers.Items[j].Spec.MinReplicas = minReplicas
```

Comment:
Note to reviewers: Had to change this block to operate on the original (it was previously operating on the loop variable, which was an ephemeral copy scoped to this block) so we can detect maxReplicas==0 below in the deletion discovery path.